### PR TITLE
filecoin-proofs re-organization

### PR DESCRIPTION
**Pull request by cryptonemo**

### CODE REVIEW

File: filecoin-proofs/src/fr32.rs [hunk 1]
```diff
@@ -855,179 +855,3 @@ where
 
     Ok(raw_data.len())
 }
-
-#[cfg(test)]
```
Comment (dignifiedquire) [inline filecoin-proofs/src/fr32.rs:859]: lets keep these tests in here, they are quite simple and unity

File: filecoin-proofs/tests/constants.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+use filecoin_proofs::constants::*;
+use filecoin_proofs::with_shape;
+
+#[cfg(test)]
+mod tests {
```
Comment (dignifiedquire) [inline filecoin-proofs/tests/constants.rs:5]: no need for this `#[cfg(test)]; mod tests` in the `tests` directory

File: filecoin-proofs/tests/parameters.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#[cfg(test)]
+mod tests {
+    use filecoin_proofs::constants::*;
+    use filecoin_proofs::parameters::{select_challenges, winning_post_public_params};
+    use filecoin_proofs::types::{PoStConfig, PoStType};
+
+    #[test]
```
Comment (dignifiedquire) [inline filecoin-proofs/tests/parameters.rs:7]: these are very simple unit level tests, lets keep them where they were

File: fil-proofs-param/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+version = "0.1.0"
+authors = ["dignifiedquire <dignifiedquire@gmail.com>", "laser <l@s3r.com>", "porcuquine <porcuquine@users.noreply.github.com>"]
+license = "MIT OR Apache-2.0"
+edition = "2018"
+repository = "https://github.com/filecoin-project/rust-fil-proofs"
+readme = "README.md"
+
+[dependencies]
+storage-proofs = { path = "../storage-proofs", default-features = false }
+filecoin-proofs = { path = "../filecoin-proofs", default-features = false }
```
Comment (dignifiedquire) [inline fil-proofs-param/Cargo.toml:13]: these should have a version too

**Comment by dignifiedquire** [comment]
there are a bunch of tests for the binaries in `filecoin-proofs/tests` which will need to be moved too

**Comment by cryptonemo** [comment]
Resolves #1344 and partially addresses #1349 (regarding filecoin-proofs only)

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**